e2e/wdio/xvfb/base-install.e2e.ts

```diff
@@ -132,18 +158,240 @@ describe('xvfb fresh installation', () => {
         })
     })
 
+    describe('advanced autoInstall configurations', () => {
+        it('should handle object format with mode specification', async function(this: Mocha.Context) {
+            this.timeout(300000) // 5 minutes
+
+            // Test object format with explicit sudo mode
+            const manager = new XvfbManager({
+                autoInstall: { mode: 'sudo' }
+            })
+
+            delete process.env.DISPLAY
+            expect(manager.shouldRun()).toBe(true)
+
+            const result = await manager.init()
+            expect(result).toBe(true)
+        })
+
+        it('should support custom install commands', async function(this: Mocha.Context) {
+            this.timeout(300000) // 5 minutes
+
+            const testMarkerFile = '/tmp/xvfb-custom-command-test'
+
+            // Clean up any existing marker file
+            try {
+                execSync(`rm -f ${testMarkerFile}`, { stdio: 'ignore' })
```

Comment:
Seems like we are using real files rather than mocks, if this is the case, can we use mocks instead?